analysis/Scripts/Efficiencies/efficiencies.py

```diff
@@ -0,0 +1,191 @@
+from ROOT import TChain, TCanvas, TH1, TFile
+import ROOT, os, Imports, sys, getopt
+from Imports import TUPLE_PATH, RAW_TUPLE_PATH, MC_jobs_Dict
+import pprint
+
+ybins = Imports.getYbins()
+ptbins = Imports.getPTbins()
+
+def main(argv):
+
+	ROOT.gROOT.SetBatch(True) #STOP SHOWING THE GRAPH FOR ROOT
+
+	try:
+		opts, args = getopt.getopt(argv,"hstp")
+	except getopt.GetoptError:
+		print("The arguments are wrong")
+		sys.exit(2)
+
+	options = []
+	arguments = []
+
+	for opt,arg in opts:
+		options.append(opt)
+		arguments.append(arg)
+
+	if not options:
+		options = ["-s","-t","-p"]
+
+	if "-h" in options:
+		print(textwrap.dedent("""\
+
+			Welcome to the efficiencies.py script.
+
+			The parameters are
+				-h : help
+				-s : Selection
+				-t : Trigger
+				-p : PID
+
+			Running with no parameter will output all the data at once.
+			"""))
+
+		sys.exit()
+
+
+	for opt in options:
+
+		if opt == "-s":
+
+			selecEffDict = {}
+
+			print("\nCreation of the selection efficiency files")
+
+			f_text = open("Selection_Eff_output.txt", "w+")
+
+			n = len(MC_jobs_Dict)
+			i = 0
+
+			for job in MC_jobs_Dict:
+				#FOR THE PROGRESSION BAR
+				if i < n:
+					j = (i + 1) / n
+					sys.stdout.write('\r')
+					sys.stdout.write("[%-20s] %d%%" % ('='*int(20*j), 100*j))
+					sys.stdout.flush()
+					i += 1
+				if job == "NA":
+					continue
+
+				particle = MC_jobs_Dict[job][3]
+				year = MC_jobs_Dict[job][0]
+				pol = MC_jobs_Dict[job][1]
+				subjobs = MC_jobs_Dict[job][2]
+				identifier = MC_jobs_Dict[job][4]
+
+				filename = "MC_Lc2pKpiTuple_" + identifier + ".root"
+
+				if int(year) <= 2012:
+					run = 1
+					cuts = Imports.getDataCuts(run)
+				else:
+					run = 2
+					cuts = Imports.getDataCuts(run)
+
+				# # APPARENTLY THAT THE ONLY CUTS THAT SIMON WAS LOOKING AT ---- !!!QUESTION!!!
+				# cuts = "lcplus_P < 300000 && lcplus_OWNPV_CHI2 < 80 &&  pplus_P < 120000 && kminus_P < 115000 && piplus_P < 80000"
+
+				# #turbo = "lcplus_Hlt2CharmHadXicpToPpKmPipTurboDecision_TOS==1"
+				# turbo = "lcplus_Hlt2CharmHadD2HHHDecision_TOS == 1"
+
+				Lc_MC_tree = TChain("tuple_Lc2pKpi/DecayTree") # !!! QUESTION : NOT BETTER ISTEAD OF CHAIN; JUST GETENTRIES FROM EACH ONE BY ONE, ONCE WITHOUT CUT AND ONCE WITH?
```

Comment:
sure we could do this, if we wanted, i don't know if that saves computer time

Reply:
Doesn't matter much. Would keep this to avoid complexity.